File: Solution/Medium/0881 Boats to Save People.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int count = 0;
        int i = 0, j = people.length - 1;
        Arrays.sort(people);
        while (i <= j) {
            if (people[j] + people[i] <= limit) {
                i++;
            }
            j--;
            count++;
        }
        return count;
    }
}

File: Solution/Medium/0881 Boats to Save People.kt
class Solution {
    fun numRescueBoats(people: IntArray, limit: Int): Int {
        people.sort()
        var res = 0
        var start = 0
        var end = people.size - 1
        while (start <= end) {
            if (people[end] + people[start] > limit) {
                end--
            } else {
                end--
                start++
            }
            res++
        }
        return res
    }
}
